src/java.base/share/classes/jdk/internal/module/ModulePath.java

```diff
@@ -551,7 +551,7 @@ private ModuleDescriptor deriveModuleDescriptor(JarFile jf)
                     if (!cn.isEmpty()) {
                         String pn = packageName(cn);
                         if (!packages.contains(pn)) {
-                            String msg = "Provider class " + cn + " not in module";
+                            String msg = "Provider class " + cn + " not in module created for " + fn;
```

Comment:
"not in module created" isn't quite right as the module cannot be created.  Maybe you could change it to "not in JAR file"

Reply:
Will do so.